### PR TITLE
[ST-Engine][PhysicalPlan] Update PhysicalPlan to support scheduler by pipeline

### PR DESCRIPTION
**Pull request by EricJoy2048**
sub task of https://github.com/apache/incubator-seatunnel/issues/2339

### CODE REVIEW

File: seatunnel-core/seatunnel-seatunnel-starter/src/main/bin/start-seatunnel-seatunnel.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  else
+    # shellcheck disable=SC2006
+    PRG=`dirname "$PRG"`/"$link"
+  fi
+done
+
+PRG_DIR=`dirname "$PRG"`
+APP_DIR=`cd "$PRG_DIR/.." >/dev/null; pwd`
+CONF_DIR=${APP_DIR}/config
+APP_JAR=${APP_DIR}/lib/seatunnel-seatunnel-starter.jar
```
Comment (Hisoka-X) [inline seatunnel-core/seatunnel-seatunnel-starter/src/main/bin/start-seatunnel-seatunnel.sh:39]: Use `seatunnel-seatunnel-starter.jar` as jar name, a little strange

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/PhysicalPlan.java [hunk 2]
```diff
@@ -17,38 +17,134 @@
 
 package org.apache.seatunnel.engine.server.dag.physical;
 
-import org.apache.seatunnel.engine.server.task.TaskGroupInfo;
+import org.apache.seatunnel.engine.common.utils.NonCompletableFuture;
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/PhysicalPlan.java:20]: I have a little doubt, why put the scheduling-related logic directly in the physical plan, why not create a new class to do this?

Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/PhysicalPlan.java:20]: > I have a little doubt, why put the scheduling-related logic directly in the physical plan, why not create a new class to do this?

I refer to Flink, spark and other computing engines, which manage the job state in PyhsicalPlan. I understand that this is reasonable. The physical execution plan is not only used to represent a DAG relationship. The execution plan should be responsible for job execution and state manager.

File: seatunnel-core/seatunnel-seatunnel-starter/src/main/bin/start-seatunnel-seatunnel.sh (hunk 1 above)
Comment (EricJoy2048) [reply seatunnel-core/seatunnel-seatunnel-starter/src/main/bin/start-seatunnel-seatunnel.sh:39]: > Use `seatunnel-seatunnel-starter.jar` as jar name, a little strange

Yes, a little. We can discuss the module name in the feature.

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/PhysicalPlan.java (hunk 2 above)
Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/PhysicalPlan.java:20]: Got it

**Pull request merged**